Lib/test/test_importlib/identity.py

```diff
@@ -0,0 +1,404 @@
+import re
```

Comment:
The hypothesis strategy is really clunky and complex, isn't it? But it did identify some edge cases that required me to rewrite both regexes, if you look at my commit history. I think that's justification enough. And believe it or not, the regexes are written to be simple and straightforward to understand. There's a lot of possible optimizations to the regexes, such as loop unrolling, for which having this hypothesis strategy might eventually prove more useful.

Reply:
Yeah, I guess that's a good justification. I do appreciate that the tests are noticeably enhanced by the adoption of hypothesis. Whether or not we split out the hypothesis concern, I'd definitely want to be sure it can be adopted upstream in importlib_metadata.

Reply:
I'll separate out the Hypothesis identities strategy as a separate PR for importlib_metadata.